plugins/repository-azure/src/test/java/org/elasticsearch/cloud/azure/storage/AzureStorageServiceMock.java

```diff
@@ -95,13 +95,13 @@ public OutputStream getOutputStream(String account, LocationMode mode, String co
         MapBuilder<String, BlobMetaData> blobsBuilder = MapBuilder.newMapBuilder();
         for (String blobName : blobs.keySet()) {
             final String checkBlob;
-            if (keyPath != null) {
+            if (keyPath != null || keyPath.isEmpty()) {
```

Comment:
I don't follow this logic, don't you want `keyPath != null && !keyPath.isEmpty())`?